internal/engine/synclet_build_and_deployer.go

```diff
@@ -26,36 +26,59 @@ type SyncletBuildAndDeployer struct {
 
 	kCli k8s.Client
 
-	deployInfo   map[docker.ImgNameAndTag]DeployInfo
+	deployInfo   map[docker.ImgNameAndTag]*DeployInfo
 	deployInfoMu sync.Mutex
 }
 
 type DeployInfo struct {
 	podID       k8s.PodID
 	containerID k8s.ContainerID
 	nodeID      k8s.NodeID
+
+	ready chan struct{} // Close this channel when the DeployInfo is populated
+}
+
+// ~~~ should these verify that the channel is non-nil?
+func (di *DeployInfo) waitUntilReady() { <-di.ready }
```

Comment:
in practice, you'll want to do something like
select {
  case <-di.ready:
  case <-ctx.Done():
}

Reply:
will do.

do i need to verify that this channel is non-nil before acting on it? or can i assume that this method is only getting called on a properly-formed `di`?

Reply:
nope! don't worry about that